### PR TITLE
Removing bosh property related operations from codebase

### PR DESCRIPTION
**Pull request by jayeshjk**
* All calls to bosh deployment properties API are being removed in this PR.
* Following are the changes:
1. Modification in `getCredentials` during unbind operation of director and virtualhost resources. Earlier provision of fallback to bosh deployment property is removed and we are now completely dependent on ApiServer to obtain credentials in unbind flow.
2. Modification in `DBManager` initialization flow to be dependent on ApiServer only.
3. Removal of various deployment property API calls from `BoshDirectorClient`
4. Corresponding changes in UTs.

### CODE REVIEW

File: data-access-layer/db/DBManager/DBManager.js [hunk 1]
```diff
@@ -118,26 +117,7 @@ class DBManager {
...
-            .then(bindInfo => {
-              return this.storeBindPropertyOnApiServer({
-                id: _.toLower(CONST.FABRIK_INTERNAL_MONGO_DB.BINDING_ID),
-                parameters: bindInfo.parameters,
-                credentials: bindInfo.credentials
-              })
-                .then(() => bindInfo);
-            });
-        });
+        .then(resource => utils.decodeBase64(_.get(resource, 'status.response')));
```
Comment (subhankarc) [inline data-access-layer/db/DBManager/DBManager.js]: Please validate that the corresponding apiserver resource is present in higher landscapes, also in new landscapes.

Comment (subhankarc) [reply data-access-layer/db/DBManager/DBManager.js]: 1. Test the re-creation with deleting sf-event-mesh-server and sf deployment and re-creating them.

File: operators/bosh-operator/DirectorService.js [hunk 2]
```diff
@@ -861,42 +846,28 @@ class DirectorService extends BaseDirectorService {
...
-      .getDeploymentProperty(deploymentName, `binding-${id}`)
-      .then(result => JSON.parse(result))
-      .catchThrow(NotFound, new ServiceBindingNotFound(id));
+      .then(response => {
+          if (response) {
+            return utils.decodeBase64(response);
+          }
+        })
+    }, {
+      maxAttempts: 5,
```
Comment (subhankarc) [inline operators/bosh-operator/DirectorService.js]: maxAttempts can be 3, delay is in milisecs ?

File: data-access-layer/db/DBManager/DBManager.js (hunk 1 above)
Comment (jayeshjk) [reply data-access-layer/db/DBManager/DBManager.js]: Both done.

**Comment by jayeshjk** [comment]
Pipeline is completely green for Azure, GCP, SC7 and Ali, but deploy is failing Aws, hence the `DO_NOT_MERGE` label. There could be minor change in `DBManager` (very unlikely). Apart from it, PR is ready for review.

File: data-access-layer/db/DBManager/DBManager.js [hunk 3]
```diff
@@ -69,8 +69,7 @@ class DBManager {
         logger.info(`Connecting to DB with the provided config URL : ${config.mongodb.url}`);
         return this.initDb(config.mongodb);
       }
-    }) // On deterministic error, just log message and stop
-      .catch(ServiceBindingNotFound, err => logger.warn('MongoDB binding to ServiceFabrik not found. This generally should not occur. More Info:', err))
```
Comment (abh1kg) [inline data-access-layer/db/DBManager/DBManager.js]: Is this related to this change?

Comment (jayeshjk) [reply data-access-layer/db/DBManager/DBManager.js]: Earlier `ServiceBindingNotFound` was thrown in `getBindingProperty` at [LN 127](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/665/files#diff-637f9dd65cf53fe8e7cbf23bd6469127L127) below. Now we use `NotFound` from ApiServer instead. This scenario will mainly come in picture at the time of landscape recreation/scratch setup.

File: operators/bosh-operator/DirectorService.js [hunk 4]
```diff
@@ -56,27 +55,12 @@ class DirectorService extends BaseDirectorService {
...
-             */
-            logger.warn(`Deployment property '${CONST.PLATFORM_CONTEXT_KEY}' not found for instance '${this.guid}'.\ 
-          Setting default platform as '${CONST.PLATFORM.CF}'`);
-
-            const context = {
-              platform: CONST.PLATFORM.CF
-            };
-            return context;
-          });
+        if (!context) {
```
Comment (amitmalav) [inline operators/bosh-operator/DirectorService.js]: use `_.isEmpty(context)`

File: operators/virtualhost-operator/VirtualHostService.js [hunk 5]
```diff
@@ -127,28 +125,35 @@ class VirtualHostService extends BaseService {
...
+        resourceId: id
+      })
+        .then(response => {
+          if (response) {
+            return utils.decodeBase64(response);
+          }
+        });
+    }, {
+      maxAttempts: 5,
+      minDelay: 1000
```
Comment (amitmalav) [inline operators/virtualhost-operator/VirtualHostService.js]: `predicate: err => !(err instanceof NotFound)`
